### PR TITLE
Upgrade liquibase to 3.5.x version TRUNK-4986

### PR DESCRIPTION
**Pull request by georgejunior**
## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-4986

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [ x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [ x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [ ] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [ x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by HerbertYiga** [comment]
some failing checks here! does this pass locally?

**Comment by georgejunior** [comment]
> some failing checks here! does this pass locally?

no, it fails locally

File: api/src/main/java/org/openmrs/util/DatabaseUpdater.java [hunk 1]
```diff
@@ -199,12 +199,11 @@ public void visit(ChangeSet changeSet, DatabaseChangeLog databaseChangeLog, Data
...
 		Liquibase liquibase = getLiquibase(changeLogFile, cl);
 		int numChangeSetsToRun = liquibase.listUnrunChangeSets(contexts).size();
-		Database database = null;
-		LockService lockHandler = null;
+		Database database = null;. lockHandler = null;
 		
 		try {
 			database = liquibase.getDatabase();
-			lockHandler = LockService.getInstance(database);
+			lockHandler = LockServiceFactory.getInstance().getLockService;
```
Comment (dilanthas) [inline api/src/main/java/org/openmrs/util/DatabaseUpdater.java]: Is this compiling ? i believe **getLockService** is a method

Comment (georgejunior) [reply api/src/main/java/org/openmrs/util/DatabaseUpdater.java]: it isn't compiling

**Comment by georgejunior** [comment]
@dkayiwa  what do we do with this?

File: api/src/main/java/org/openmrs/util/DatabaseUpdater.java [hunk 2]
```diff
@@ -199,12 +199,11 @@ public void visit(ChangeSet changeSet, DatabaseChangeLog databaseChangeLog, Data
 		log.debug("Setting up liquibase object to run changelog: " + changeLogFile);
 		Liquibase liquibase = getLiquibase(changeLogFile, cl);
 		int numChangeSetsToRun = liquibase.listUnrunChangeSets(contexts).size();
-		Database database = null;
-		LockService lockHandler = null;
+		Database database = null;. lockHandler = null;
```
Comment (dilanthas) [inline api/src/main/java/org/openmrs/util/DatabaseUpdater.java]: @georgejunior You have a compilation issue in this line.  you need to remove **. lockHandler = null;** and add LockService lockHandler = null;

File: api/src/main/java/org/openmrs/util/DatabaseUpdater.java (hunk 1 above)
Comment (dilanthas) [reply api/src/main/java/org/openmrs/util/DatabaseUpdater.java]: replace **lockHandler = LockServiceFactory.getInstance().getLockService;** with **lockHandler = LockServiceFactory.getInstance().getLockService();** then compile the project locally ( mvn clean install )and see whether it builds without any errors before pushing your changes

File: api/src/main/java/org/openmrs/util/DatabaseUpdater.java (hunk 2 above)
Comment (georgejunior) [reply api/src/main/java/org/openmrs/util/DatabaseUpdater.java]: okay let me try that

Comment (georgejunior) [reply api/src/main/java/org/openmrs/util/DatabaseUpdater.java]: > @georgejunior You have a compilation issue in this line. you need to remove **. lockHandler = null;** and add LockService lockHandler = null;

i am still getting a build failure

Comment (dilanthas) [reply api/src/main/java/org/openmrs/util/DatabaseUpdater.java]: @georgejunior can you please push the code so we can have a look ?

Comment (dilanthas) [reply api/src/main/java/org/openmrs/util/DatabaseUpdater.java]: @georgejunior I can still see the line that i've asked you to change. You should remove the line marked in red color and keep it as its marked by blue color.

image

**Comment by dilanthas** [comment]
@georgejunior There are more compilation errors. Please go to the travis-ci console to see them.

image

**Comment by georgejunior** [comment]
> @georgejunior There are more compilation errors. Please go to the travis-ci console to see them.
> 
> image

i can see this [error]([ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.1:compile (default-compile) on project openmrs-api: Compilation failure: Compilation failure: )

**Comment by georgejunior** [comment]
> @georgejunior There are more compilation errors. Please go to the travis-ci console to see them.
> 
> image

yes i have seen these [errors](https://pastebin.com/kFmbueZC)

**Pull request closed without merging**